pkg/generator/ingress_translator.go

```diff
@@ -144,10 +161,9 @@ func (translator *IngressTranslator) translateIngress(ctx context.Context, ingre
 				// Match the ingress' port with a port on the Service to find the target.
 				// Also find out if the target supports HTTP2.
 				var (
-					externalPort  = int32(80)
-					targetPort    = int32(80)
-					http2         = false
-					httpsPortUsed = false
```

Comment:
Why we had this one in the first place?

Reply:
AFAIK: I think because of the config.FromContextOrDefault(). In the first initialization, the default value was encryption disabled. So the service port was used to find out, that is was an encrypted service. Otherwise envoy would have been configured with http until the CMW fired and it reconfigured it again with https.